contracts/L2ApplicationStateRootProvider.sol

```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.25;
+
+/// @title L2ApplicationStateRootProvider
+/// @notice The L2ApplicationStateRootProvider stores the state root of the Cosmos application using Monomer.
+///         This is necessary for implementing withdrawals because we need a valid account proof for the
+///         L2ToL1MessagePasser via the top-level Monomer state root.
+contract L2ApplicationStateRootProvider {
```

Comment:
I've only skimmed the op-spec Withdrawals section, but it strikes me as odd that L2 state root storage / update isn't already accounted for somewhere in the OP Stack L1 contracts.

Maybe more clarity on that later today.

Reply:
The `L2OutputOracle` contract is L1's view of L2 state. This contract will be deployed on an EVM embedded in monomer in a follow-up PR. We need to store this data on monomer so we can eventually generate the account proof (`eth_getProof`) that the withdrawal tx was executed on the cosmos application.

Reply:
Gotchya.

My understanding was that Cosmos doesn't natively support solidity (seems correct?). This led me to believe this contract was for L1.

Can defer followup until this afternoon.